packages/bff/src/services/agreementService.ts

```diff
@@ -96,6 +96,41 @@ export function agreementServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return enhanceAgreementDetailed(agreement, clients, ctx);
     },
+
+    async deleteAgreement(
+      agreementId: string,
+      { headers }: WithLogger<BffAppContext>
+    ): Promise<void> {
+      return await agreementProcessClient.deleteAgreement(undefined, {
+        params: { agreementId },
+        headers,
+      });
+    },
+
+    async activateAgreement(
+      agreementId: string,
+      ctx: WithLogger<BffAppContext>
```

Comment:
```suggestion
      { header }: WithLogger<BffAppContext>
```

Reply:
Actually in that case i cannot do that because i need to pass the entire `ctx` to enhanceAgreement. Is there any way to directly have both destructured and full object in params?

Reply:
Maybe add a const { headers } = ctx; as first instruction of the function, but it's ok as it is now